src/main/java/com/joyent/manta/monitor/MantaMonitorJerseyServer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2018, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.monitor;
+
+import com.google.inject.Injector;
+import com.google.inject.servlet.GuiceFilter;
+import com.google.inject.servlet.GuiceServletContextListener;
+import io.logz.guice.jersey.GuiceJerseyResourceConfig;
+import io.logz.guice.jersey.JettyServerCreator;
+import io.logz.guice.jersey.configuration.JerseyConfiguration;
+import io.logz.guice.jersey.configuration.ServerConnectorConfiguration;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.DispatcherType;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Customized version of the {@link io.logz.guice.jersey.JerseyServer}.
+ * This class adds configuration of SSL connector which handling HTTPS.
+ */
+
+public class MantaMonitorJerseyServer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MantaMonitorJerseyServer.class);
+    private final JerseyConfiguration jerseyConfiguration;
+    private final Supplier<Injector> injectorSupplier;
+    private final Server server;
+
+    MantaMonitorJerseyServer(final JerseyConfiguration jerseyConfiguration,
```

Comment:
Typically you never need reference to an injector when creating a Guice module. I see why you have one - you need it to get an instance of a `Configuration` class.

However, I want to question why we put the port number in the configuration file. We may be better off making it an environment variable and using the configuration file strictly to define the testing behavior. This would simplify the injection behavior and make the configuration more single-themed.

Reply:
We should also do this for the secure port.

Reply:
Totally agree on making the ports as env variable.